tests/tests.rs

```diff
@@ -831,6 +831,17 @@ sherlock\x00can extract a clew from a wisp of straw or a flake of cigar ash;
     assert_eq!(lines, expected);
 });
 
+// See: https://github.com/BurntSushi/ripgrep/issues/109
+clean!(max_depth, "far", ".", |wd: WorkDir, mut cmd: Command| {
+    wd.create_dir("one");
+    wd.create_dir("one/too");
+    wd.create("one/too/many", "far");
+
+    cmd.arg("--maxdepth").arg("1");
+
+    wd.assert_err(&mut cmd);
```

Comment:
OK, so I think with `one/foo`, you do indeed need to set `maxdepth` to `2`. The argument, `.`, corresponds to depth `0`. `one` is depth `1` and `one/foo` is depth `2`.